experimental/list_formatter/src/list_formatter.rs

```diff
@@ -2,47 +2,52 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use crate::{provider::ListFormatterPatternsV1, ListStyle};
+use crate::provider::{AndListV1Marker, ErasedListV1Marker, OrListV1Marker, UnitListV1Marker};
+use crate::ListStyle;
 use core::fmt::{self, Write};
 use icu_locid::Locale;
 use icu_provider::prelude::*;
 use writeable::*;
 
 /// A formatter that renders sequences of items in an i18n-friendly way. See the
 /// [crate-level documentation](crate) for more details.
-pub struct ListFormatter<M: ResourceMarker<Yokeable = ListFormatterPatternsV1<'static>>> {
-    data: DataPayload<M>,
+pub struct ListFormatter {
+    data: DataPayload<ErasedListV1Marker>,
     style: ListStyle,
 }
 
-impl<M: ResourceMarker<Yokeable = ListFormatterPatternsV1<'static>>> ListFormatter<M> {
-    /// Creates a new [`ListFormatter`] that produces a list for the given [`ResourceMarker`]. See
-    /// [`crate::markers`].
-    pub fn try_new<T: Into<Locale>, D: ResourceProvider<M> + ?Sized>(
-        locale: T,
-        data_provider: &D,
-        style: ListStyle,
-    ) -> Result<Self, DataError> {
-        let data = data_provider
-            .load_resource(&DataRequest {
-                options: locale.into().into(),
-                metadata: Default::default(),
-            })?
-            .take_payload()?;
-        Ok(Self { data, style })
-    }
+macro_rules! constructor {
+    ($name: ident, $marker: ty, $doc: literal) => {
+        #[doc = concat!("Creates a new [`ListFormatter`] that produces a ", $doc, "-type list. See the [CLDR spec]",
```

Comment:
Praise: This `#[doc = concat!()]` trick does actually appear to work.  This is a nice option when needing to auto-generate doc strings in macros.

http://icu4x-pr-artifacts.storage.googleapis.com/gha/811a0b003fbdd211572f01b5e44d7a834355d92d/docs/icu_list/struct.ListFormatter.html